test/tic_tac_toe/core_test.clj

```diff
@@ -0,0 +1,38 @@
+(ns tic-tac-toe.core-test
+  (:require [clojure.test :refer :all]
+            [tic-tac-toe.user-interface :as ui]
+            [tic-tac-toe.io :as io]
+            [tic-tac-toe.player :as player]
+            [tic-tac-toe.core :refer :all]))
+
+(def board-3x3 [" " " " " "
+                " " " " " "
+                " " " " " "])
+(def move-added [" " " " " "
+                " " "X" " "
+                " " " " " "])
+(def board-full ["X" "X" "X"
+                 "X" "X" "X"
+                 "X" "X" "X"])
+(def game-over "Game over.")
+(def game-settings {:size 3 :board board-3x3})
+(def clear-string (format "\033[2J"))
+(def divider-3x3 "\n--- + --- + ---\n")
+(def row-3x3 " X  |  X  |  X ")
+(def board-string (str row-3x3 divider-3x3 row-3x3 divider-3x3 row-3x3 "\n\n"))
+
+(deftest sets-up-game
+  (testing "returns a hashmap of game settings"
+    (is (= game-settings (game-setup)))))
+
+(deftest takes-turn-on-board
+  (testing "gets a move form the user and adds to board"
+    (with-redefs [ui/display-board (constantly "")
+                  ui/message (constantly "")
+                  player/pick-move (constantly 4)]
+      (is (= move-added (take-turn board-3x3 3))))))
+
+(deftest handles-game-over
+  (testing "ends game if board is full"
+    (with-redefs [ui/display-board (constantly "")]
```

Comment:
`with-redefs` is like the Clojure equivalent to the Rspec mocking API. Can you refactor so that you don't have to use `with-redefs`?

Reply:
I should no longer have `with-redefs` in tests.
[Core](https://github.com/saralein/tic-tac-toe_clojure/pull/1/files#diff-1ca7190a9dadb1c54eb37222b58bb769R1)
[Game](https://github.com/saralein/tic-tac-toe_clojure/pull/1/files#diff-0d30ef7a7274daa4a3329788cba3d84fR1)